ls1intum/Artemis
# [Feature] Store ID of parent exercise in imported exercise

### Checklist

### Motivation and Context
See: #3441 
Currently we do not store any connection between an imported exercise and the one from which it was imported from. However, in order to be able to reuse the feedback of old assessments we need to establish such a connection between the parent and the imported exercises.

As discussed on #3441 this connection is established between all types of exercises that can be imported and not only between TextExercises.

### Description
When we import an exercise, the ID of the parent exercise is saved in a new column named `imported_exercise_id`. This value helps in establishing a connection between the parent exercise and the imported ones. 

Deleting an imported exercise doesn't have any effect on the parent exercise therefore they can be easily deleted. However, when we delete the parent exercise the `imported_exercise_id` is set to `null` so that we break the connection between the parent and the imported exercise.

### Steps for Testing

1. Log in to Artemis as a Tutor or Instructor
2. Navigate to Course Management
3. Open the Network panel on your browser or check database changes using any Database management tool you prefer
4. Create a new text exercise
5. Import a new text exercise from the one initially created
6. Check if `importedExerciseId` is set to the ID of the parent exercise
7. Delete the parent exercise
8. Check if `importedExerciseId` is set to `null`

### Test Coverage
[Exercise.java](https://github.com/ls1intum/Artemis/pull/3472/files#diff-7229933eed5ab0f9444d3f8be5e1982080cbdf71dfdfddc840df0e1bf42a1d62) -> 94%
[ExerciseRepository.java](https://github.com/ls1intum/Artemis/pull/3472/files#diff-1f5b21dde0a69b3af555f7ce96352bdde291221f41fb52ec9b4ae763942def8d) -> 89%
[ExerciseImportService.java](https://github.com/ls1intum/Artemis/pull/3472/files#diff-0e0fbcc05b0e3bdbeee6a692dae9d47693047e8e0fee6658d5b8204cf770549e) -> 100%
[ExerciseService.java](https://github.com/ls1intum/Artemis/pull/3472/files#diff-1b78369c3fa7f755c8c69d6774991dcb0069d3f8c74ff85dfe8e338385c9bb23) -> 91%

### Screenshots
![artemis_1](https://user-images.githubusercontent.com/17032718/119237622-0f0feb00-bb3e-11eb-8c33-5f96b50e7ee6.gif)
